proxy.go

```diff
@@ -57,14 +79,17 @@ func ProxyHandler(w http.ResponseWriter, req *http.Request) {
 		authenticated, err := http.NewRequest("GET", req.URL.String(), req.Body)
 		authenticated.Header.Set("On-Behalf-Of", user)
 		var solutionMsg string
+		var client *http.Client
```

Comment:
I think you can get rid of this line if you're assigning client to `p.HttpClient` below.